diff -up zip31d34/unix/Makefile.cflags zip31d34/unix/Makefile
--- zip31d34/unix/Makefile.cflags	2018-03-21 04:44:05.180150346 +0100
+++ zip31d34/unix/Makefile	2018-03-21 04:44:30.512138920 +0100
@@ -380,7 +380,7 @@ CC      = cc
 CC_BZ   = $(CC)
 
 CF_NOOPT = -I. -DUNIX $(LOCAL_ZIP)
-CFLAGS  = -O
+CFLAGS  =
 CF      = $(CFLAGS) $(CF_NOOPT)         # C compiler flags (core components)
 CF_BZ   = $(CF)                         # C compiler flags (BZIP2 components)
 
